### PR TITLE
[LoRA] Freezing the model weights

### PR DESCRIPTION
**Pull request by erkams**
As in original [lora repository](https://github.com/cloneofsimo/lora/blob/master/training_scripts/train_lora_pt_caption.py#L655) freezing the weights of the models helps us to use less memory

### CODE REVIEW

File: examples/text_to_image/train_text_to_image_lora.py [hunk 1]
```diff
@@ -415,7 +416,18 @@ def main():
     unet = UNet2DConditionModel.from_pretrained(
         args.pretrained_model_name_or_path, subfolder="unet", revision=args.revision
     )
-
+    unet.requires_grad_(False)
```
Comment (patrickvonplaten) [inline examples/text_to_image/train_text_to_image_lora.py:419]:
```suggestion
    # freeze parameters of models to save more memory
    unet.requires_grad_(False)
```

**Review by patrickvonplaten — APPROVED** [review]
This makes sense to me! @patil-suraj @williamberman could you also take a quick look?

File: examples/text_to_image/train_text_to_image_lora.py [hunk 2]
```diff
@@ -415,7 +416,18 @@ def main():
...
+    vae.requires_grad_(False)
+    
+    params = itertools.chain(
```
Comment (patrickvonplaten) [inline examples/text_to_image/train_text_to_image_lora.py:422]: Why not just to `text_encoder.requires_grad_(False)` ?

**Comment by patrickvonplaten** [comment]
Also cc @sayakpaul - I think we can freeze everything here just like we do for dreambooth_lora

**Comment by sayakpaul** [comment]
> Also cc @sayakpaul - I think we can freeze everything here just like we do for dreambooth_lora

+1

File: examples/text_to_image/train_text_to_image_lora.py (hunk 2 above)
Comment (patil-suraj) [reply examples/text_to_image/train_text_to_image_lora.py:422]: +1,  , this way of freezing is only needed for the textual inversion
here we can use `text_encoder.requires_grad_(False)`

File: examples/text_to_image/train_text_to_image_lora.py [hunk 3]
```diff
@@ -415,7 +416,18 @@ def main():
...
+        text_encoder.text_model.encoder.parameters(),
+        text_encoder.text_model.final_layer_norm.parameters(),
+        text_encoder.text_model.embeddings.position_embedding.parameters(),
+    )
+    
+    for param in params:
+        param.requires_grad = False
```
Comment (patil-suraj) [inline examples/text_to_image/train_text_to_image_lora.py:429]:
```suggestion
    text_encoder.requires_grad_(False)
```

**Review by patil-suraj — APPROVED** [review]
Thanks a lot for the PR, looks good, just left one comment about freezing text encoder.

File: examples/text_to_image/train_text_to_image_lora.py [hunk 4]
```diff
@@ -21,6 +21,7 @@
 import random
 from pathlib import Path
 from typing import Optional
+import itertools
```
Comment (patrickvonplaten) [inline examples/text_to_image/train_text_to_image_lora.py:24]:
```suggestion
```

**Review by pcuenca — APPROVED** [review]
Makes complete sense to me!

**Comment by patrickvonplaten** [comment]
Thanks a lot @erkams !

**Pull request merged**